### DISCUSSION TITLE
Essential files/directories for next start

### DISCUSSION THREAD

**Discussion by JuanIrache** [vercel/next.js#37070] [Help]
Hi,

I am going to run the same nextjs app from a couple of servers. A load balancer distributes traffic to both of them. One of them acts as a "master" and its contents are copied to the other one via lsyncd. My concern is the number of files I will need to keep in sync. With apps that just build the site once and can be served from a directory with Apache, I can just sync the built version, but with Nextjs I feel like I should share the entire project structure. Is that correct?

On the "non-master" server I just need to run "next start", as the "master" does the build step. Which directories and files do I need for that? I guess at least build/ public/ package.json and node_modules/.bin/next

Is the entire node_modules/ directory necessary for "next start" to run successfully? That would be problematic, given the number of files there.

I suppose src/ is not necessary.

Thank you.

**Comment by icyJoseph** [comment]
I'd try with https://nextjs.org/docs/advanced-features/output-file-tracing#automatically-copying-traced-files-experimental, for your use case. Look at this docker file, https://github.com/vercel/next.js/blob/canary/examples/with-docker/Dockerfile, which at the end starts `server.js`.

Otherwise, I'd say, copy `.next`, `public`, `package.json`, and whatever lock file you have, and install production dependencies on "non-master", with `npm ci` for example, and then run `next start`. There's a reason why they have that tracing files feature, what to copy is not so trivial, I think at least.

**Reply by JuanIrache** [reply]
Thank you for the comprehensive answer. Will give those options a try
